examples/hosts/app-integration/external-controller/src/controller.ts

```diff
@@ -29,13 +29,26 @@ export interface IDiceRollerController extends EventEmitter {
 // The data is stored in a key-value pair data object, so we'll use this key for storing the value.
 const diceValueKey = "diceValue";
 
+interface DiceRollerControllerProps {
+    get: (key: string) => any;
+    set: (key: string, value: any) => void;
+    on(event: "changed" | "valueChanged", listener: (args: IDirectoryValueChanged) => void): this;
+    off(event: "changed" | "valueChanged", listener: (args: IDirectoryValueChanged) => void): this;
+}
+
 /**
  * The DiceRoller is our data object that implements the IDiceRoller interface.
  */
 export class DiceRollerController extends EventEmitter implements IDiceRollerController {
-    constructor(private readonly kvPairDataObject: IKeyValueDataObject) {
+    constructor(private readonly props: DiceRollerControllerProps) {
         super();
-        this.kvPairDataObject.on("changed", (changed) => {
+        this.props.on("changed", (changed) => {
+            if (changed.key === diceValueKey) {
+                // When we see the dice value change, we'll emit the diceRolled event we specified in our interface.
+                this.emit("diceRolled");
+            }
+        });
+        this.props.on("valueChanged", (changed) => {
```

Comment:
It looks like this is to support how `KeyValueDataObject` and `SharedMap` have different event names but it looks so weird here.  Wouldn't need to happen here but is there a problem with using `valueChanged` for the event name in `KeyValueDataObject`?

Reply:
I 100% agree it's terrible. We should the future of KeyValueDataObject after this change is in.